pkg/transport/generic_transport_controller.go

```diff
@@ -827,43 +831,38 @@ func (c *genericTransportController) computeDestToCustomizedObjects(objectsToPro
 					objC = objToPropagate
 				}
 			}
-			if customizeThisObject && destToCustomizedObjects == nil {
-				destToCustomizedObjects = map[v1alpha1.Destination][]*unstructured.Unstructured{}
+			if customizeThisObject && destToCustomizedWrapees == nil {
+				destToCustomizedWrapees = map[v1alpha1.Destination][]Wrapee{}
 				for _, dest := range binding.Spec.Destinations {
-					destToCustomizedObjects[dest] = abstract.SliceCopy(objectsToPropagate[:objIdx])
+					destToCustomizedWrapees[dest] = abstract.SliceCopy(uncustomizedWrapees[:objIdx])
 				}
 			}
-			if destToCustomizedObjects != nil {
-				customizedObjectsSoFar := destToCustomizedObjects[dest]
-				customizedObjectsSoFar = append(customizedObjectsSoFar, objC)
-				destToCustomizedObjects[dest] = customizedObjectsSoFar
+			if destToCustomizedWrapees != nil {
+				customizedObjectsSoFar := destToCustomizedWrapees[dest]
+				customizedObjectsSoFar = append(customizedObjectsSoFar, Wrapee{objC, createOnly})
+				destToCustomizedWrapees[dest] = customizedObjectsSoFar
 			}
 		}
 	}
 	// update the index in c.bindingSensitiveDestinations
 	var cares sets.Set[v1alpha1.Destination]
-	if destToCustomizedObjects != nil {
+	if destToCustomizedWrapees != nil {
 		cares = sets.New(binding.Spec.Destinations...)
 	} else {
 		cares = sets.New[v1alpha1.Destination]()
 	}
 	c.setBindingSensitivities(binding.Name, cares) // forget about now-irrelevant destinations
 
-	return destToCustomizedObjects, bindingErrors
-}
-
-// TODO: replace all usage of this function with something that actually copes with the create-only bit.
-func FIXME_ADD_CONSTANT_CREATEONLY(obj *unstructured.Unstructured) Wrapee {
-	return Wrapee{obj, false}
+	return destToCustomizedWrapees, bindingErrors
 }
 
-func (c *genericTransportController) wrapBatch(batchToPropagate []*unstructured.Unstructured, binding *v1alpha1.Binding, numShard int, isSharded bool) (*unstructured.Unstructured, error) {
+func (c *genericTransportController) wrapBatch(batchToPropagate []Wrapee, binding *v1alpha1.Binding, numShard int, isSharded bool) (*unstructured.Unstructured, error) {
 	var wrapped runtime.Object
 	if t2, is := c.transport.(TransportWithCreateOnly); is {
-		wrapees := abstract.SliceMap(batchToPropagate, FIXME_ADD_CONSTANT_CREATEONLY)
-		wrapped = t2.WrapObjectsHavingCreateOnly(wrapees)
+		wrapped = t2.WrapObjectsHavingCreateOnly(batchToPropagate)
```

Comment:
Something is not clear to me in the logic. A binding may now contain objects with `CreateOnly==true` and objects with false. Does this not mean that bindings may now lead to the generation of 2 (or more) wrapped-objects? That is putting sharding aside.

Apart from this point, everything looks good.

Reply:
I could not see an explicit answer in the referenced doc or API documentation clause. I can only see the desired behavior from setting `CreateOnly` bit which I agree with, therefore I assume that the point will be clearer in upcoming PRs.

Reply:
Then I am not sure I understand your question. Is it answered by saying that each BindingPolicy object has exactly one corresponding Binding object, in which everything said about a given workload object is combined into one ClusterScopeDownsyncClause or NamespaceScopeDownsyncClause? You already knew that and implemented the StatusCollector aspect of that.

Reply:
I was under the impression that to handle the requirement of create-only ManifestWorks we'd need to split selected objects (of a binding) into two groups: one that are wrapped in a possibly sharded manifestwork as we've done to this point, and another that are wrapped in a possibly sharded create-only manifestwork. I do not see the implementation of this separation in this PR, or the groundwork for it.

If my assumption is wrong, please let me know.

Reply:
In a ManifestWork each workload object can have its own UpdateStrategy (in the ManifestConfigs of the ManifestWorkSpec).

Reply:
But, more to the point, this is a detail in the transport plugin. The interface to the transport plugin passes the needed information, a create-only bit per workload object.

Reply:
I see, thanks.